data-tool/flows/common/corp_processing_queue_service.py

```diff
@@ -0,0 +1,173 @@
+                result = conn.execute(
+                    text(init_query),
+                    {
+                        'flow_name': self.flow_name,
+                        'status': ProcessingStatuses.PENDING,
+                        'environment': self.data_load_env,
+                        'flow_run_id': flow_run_id
+                    }
+                )
+                count = len(result.fetchall())
+                self.logger.info(f"Initialized {count} corps for flow run {flow_run_id}")
+                return count
+
+    def claim_batch(self, flow_run_id: str, batch_size: int) -> List[str]:
+        """Claim a batch of corporations for immediate processing within a flow run.
+
+        Args:
+            flow_run_id: Unique identifier for this flow run
+            batch_size: Maximum number of corps to claim
+
+        Returns:
+            List of corporation numbers that were successfully claimed for processing
+        """
+        query = """
+        WITH claimable AS (
+            SELECT corp_num
+            FROM corp_processing
+            WHERE processed_status = :pending_status
+            AND environment = :environment
+            AND flow_name = :flow_name
+            AND flow_run_id = :flow_run_id
+            AND claimed_at IS NULL
+            LIMIT :batch_size
+            FOR UPDATE SKIP LOCKED
+        )
+        UPDATE corp_processing
+        SET processed_status = :processing_status,
+            claimed_at = NOW(),
+            last_modified = NOW()
+        FROM claimable
+        WHERE corp_processing.corp_num = claimable.corp_num
+        RETURNING corp_processing.corp_num, corp_processing.claimed_at
+        """
+
+        with self.db_engine.connect() as conn:
+            with conn.begin():
+                result = conn.execute(
+                    text(query),
+                    {
+                        'pending_status': ProcessingStatuses.PENDING,
+                        'processing_status': ProcessingStatuses.PROCESSING,
+                        'environment': self.data_load_env,
+                        'flow_name': self.flow_name,
+                        'flow_run_id': flow_run_id,
+                        'batch_size': batch_size
+                    }
+                )
+                claimed = result.fetchall()
+                claimed_corps = [row[0] for row in claimed]
+                if claimed_corps:
+                    self.logger.info(f"Claimed {len(claimed_corps)} corps for flow {flow_run_id}")
+                    self.logger.info(f"Corps: {', '.join(claimed_corps[:5])}...")
+                    for corp, claimed_at in claimed:
+                        self.logger.debug(f"  {corp} claimed at {claimed_at}")
+                return claimed_corps
+
+    def update_corp_status(
+        self,
+        flow_run_id: str,
+        corp_num: str,
+        status: ProcessingStatuses,
+        error: str = None
+    ) -> bool:
+        """Update status for a corp."""
```

Comment:
will make this more flexible and take in dynamic properties as we flesh this out in future work